frog/imports/api/users.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+import { Meteor } from 'meteor/meteor';
+
+type User = {
+  id?: string,
+  userObj?: mixed
+};
+
+/**
+ * Returns the appropriate username based on the type of user. If no user is passed then will return the current user's username
+ *
+ * @param: {User=} user
+ */
+
+export const getUsername = (user: User) => {
+  const selectedUser = getUser(user);
+  if (selectedUser) {
+    if (selectedUser.isAnonymous) return 'Anonymous User';
+    else if (selectedUser.emails) return selectedUser.profile.displayName;
+    else if (selectedUser.username) return selectedUser.username;
+  }
+  return 'No user logged in';
+};
+
+/**
+ * Returns the appropriate user object based on the type of user. If no user is passed as args then will return the current user object.
+ *
+ * @param: {User=} user
+ */
+const getUser = (user: User) => {
+  if (!user) return Meteor.user();
+  else {
+    const { id, userObj } = user;
+    return Meteor.users.findOne(id) || userObj;
```

Comment:
This is a minor issue, but if id is undefined, better to avoid a database lookup? Maybe

`if(id) { return Meteor.users.findOne(id) } else if userObj { return userObj } else { return Meteor.user() } `